pyiron_base/project/maintenance.py

```diff
@@ -2,14 +2,48 @@
 import os
 import pkgutil
 import warnings
+import sys
+import importlib
 
 import pandas
 
 from pyiron_base import state
 from pyiron_base.database.performance import get_database_statistics
+import pyiron_base.storage.hdfio
 from pyiron_base.project.update.pyiron_base_03x_to_04x import pyiron_base_03x_to_04x
 
 
+# we sometimes move classes between modules; this would break HDF storage,
+# since objects save there the module path from which their classes can be
+# imported.  We can work around this by defining here an explicit map that
+# _to_object can use to find the new modules and update the HDF5 files
+_MODULE_CONVERSION_DICT = {
+    "pyiron_base.generic.datacontainer": "pyiron_base.storage.datacontainer",
+    "pyiron_base.generic.inputlist": "pyiron_base.storage.inputlist",
+    "pyiron_base.generic.flattenedstorage": "pyiron_base.storage.flattenedstorage",
+    "pyiron_base.table.datamining": "pyiron_base.jobs.datamining",
+}
+
+
+def add_module_conversion(old: str, new: str):
+    """
+    Add a new module conversion.
+    After this call any object that was previously defined in module `old` and is now defined in `new`, will be
```

Comment:
This is not strictly true, or? It looks to me like you need to invoke `add_module_conversion` then still do a maintenance cycle.

Reply:
Correct, I had just copied this over from #1234, my bad.